packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -35,9 +48,18 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
     // Use CryptoWatch's most granular option, one minute.
     this.ohlcPeriod = 60;
 
-    // Helper functions from web3.
-    this.toBN = this.web3.utils.toBN;
-    this.toWei = this.web3.utils.toWei;
+    // Utility conversion, from eth (10e18) to arbitrary decimals
+    // This will become the output for all numbers in this class
+    const converter = ConvertDecimals(18, decimals);
+
+    // Crypto watch returns a number in "eth" which gets converted to wei (10**18).
+    // We need to convert this to the natural value of the collateral,
+    // which is passed in as "decimals" (10**decimals).
+    this.convertDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return converter(this.web3.utils.toWei(number.toString()));
```

Comment:
Instead of going from number --> 18 decimals --> desired decimals, why not just go straight from number --> desired decimals via the `parseFixed` module from [@ethersproject/bignumber](https://github.com/ethers-io/ethers.js/blob/master/packages/units/src.ts/index.ts#L64)?

`ethers.js` uses it in its ethers units helper classes, [docs](https://github.com/ethers-io/ethers.js/blob/master/packages/units/src.ts/index.ts#L64)

And I use it in my [PR](https://github.com/UMAprotocol/protocol/pull/1899) as another example

Reply:
i am going to try that